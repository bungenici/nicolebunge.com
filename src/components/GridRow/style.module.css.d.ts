/* eslint-disable @typescript-eslint/camelcase */
/* eslint-disable import/prefer-default-export */

export const grid__row: string;
