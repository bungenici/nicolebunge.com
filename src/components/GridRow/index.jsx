import PropTypes from 'prop-types';
import React, { PureComponent } from 'react';
import styles from './style.module.css';

class GridRow extends PureComponent {
  static propTypes = {
    children: PropTypes.node,
  };

  static defaultProps = {
    children: null,
  };

  render() {
    const { children } = this.props;

    return (
      <div className={styles.grid__row}>
        {children}
      </div>
    );
  }
}

export default GridRow;
