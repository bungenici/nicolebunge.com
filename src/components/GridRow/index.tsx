import classNames from 'classnames';
import React from 'react';
import styles from './style.module.css';

export interface GridRowProps {
  className?: string;
}

const GridRow: React.FC<GridRowProps> = (props) => {
  const { children, className, ...otherProps } = props;

  return (
    <div className={classNames(className, styles.grid__row)} {...otherProps}>
      {children}
    </div>
  );
};

export default GridRow;
