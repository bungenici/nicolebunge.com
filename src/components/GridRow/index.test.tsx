/* eslint-disable import/no-extraneous-dependencies */

import React from 'react';
import renderer from 'react-test-renderer';
import GridRow from '.';

describe('GridRow component', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<GridRow />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
