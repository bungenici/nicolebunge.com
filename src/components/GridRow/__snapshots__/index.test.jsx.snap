// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GridRow component renders correctly 1`] = `
<div
  className="grid__row"
/>
`;
